app/screens/channel/channel_title/index.js

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {connect} from 'react-redux';
+
+import {getCurrentChannel} from 'mattermost-redux/selectors/entities/channels';
+
+import {getTheme} from 'app/selectors/preferences';
+
+import ChannelTitle from './channel_title';
+
+function mapStateToProps(state) {
+    const currentChannel = getCurrentChannel(state);
+
+    return {
+        currentChannelName: currentChannel ? currentChannel.display_name : '',
+        displayName: state.views.channel.displayName,
+        theme: getTheme(state)
+    };
+}
+
+function areStatesEqual(next, prev) {
```

Comment:
Is this needed? The component will only rerender if name or theme changes, and the mapStateToProps isn't complex since getCurrentChannel is memoized properly

Reply:
will change this back as there is not really an important gain by avoiding re-renders on this one, yet you can argue that re-render + re-render elsewhere will make the difference, thoughts?

Reply:
I don't think it'll rerender though? It'll call mapStateToProps, but the ChannelTitle will see that the theme and channel names haven't changed and not rerender